exchanges/exchange.go

```diff
@@ -1516,9 +1516,22 @@ func (b *Base) GetKlineRequest(pair currency.Pair, a asset.Item, interval kline.
 		return nil, err
 	}
 
-	// NOTE: This check is here to make sure a client is notified that using
-	// this functionality will result in error if the total candles cannot be
+	// NOTE: The checks below make sure a client is notified that using this
+	// functionality will result in error if the total candles cannot be
 	// theoretically retrieved.
+	if setTimeWindow {
+		// setTimeWindow means that the end point has a fixed candle return
+		// amount from time.Now(). e.g. 300 hourly candles, so start time cannot
+		// be more than 300 hours into the past.
+		count := kline.TotalCandlesPerInterval(start, time.Now(), exchangeInterval)
+		if count > int64(b.Features.Enabled.Kline.ResultLimit) {
+			return nil, fmt.Errorf("candles count exceeded: %d endpoint has a set candle limit return: %d from time.Now(), candle data will be incomplete %w",
```

Comment:
Additionally, would you consider returning what the last point of data a requester can reach? Say I call get candles from the CLI and get this response, it would elimintate my guesswork on the startdate